modules/client/src/internal/client/methods.ts

```diff
@@ -1075,4 +1084,69 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     }
     return version;
   }
+
+  public async isDaoUpdateProposalValid(
+    proposalId: string,
+  ): Promise<DaoUpdateProposalValidity> {
+    if (!isProposalId(proposalId)) {
+      throw new InvalidProposalIdError();
+    }
+    // search for the proposal
+    const name = "IProposal";
+    const query = QueryIProposal;
+    type T = { iproposal: SubgraphIProposal };
+    const { iproposal } = await this.graphql.request<T>({
+      query,
+      params: { id: proposalId.toLowerCase() },
+      name,
+    });
+    if (!iproposal) {
+      throw new ProposalNotFoundError();
+    }
+    const res: DaoUpdateProposalValidity = {
+      version: false,
+      implementation: false,
+      initData: false,
+      actions: false,
+    };
+    // check if the proposal has only one action for upgrading the dao
+    if (iproposal.actions.length === 1) {
+      const action = iproposal.actions[0];
+      try {
+        const upgradeToAdCallParams = decodeUpgradeToAndCallAction(
+          hexToBytes(action.data),
+        );
+        const decodedInitializeFromParams = decodeInitializeFromAction(
+          upgradeToAdCallParams.data,
+        );
+        // action is decoded that mean the actions are valid
+        res.actions = true
+        const currentDaoVersion = await this.getProtocolVersion(
+          iproposal.dao.id,
+        );
+        // check if the version of the specified dao matches the version in the encoded data
+        if (
+          JSON.stringify(currentDaoVersion) ===
+            JSON.stringify(decodedInitializeFromParams.previousVersion)
+        ) {
+          res.version = true;
+        }
+        // check if the implementation matches the implementation in the latest daoFactoryAddress
+        if (
+          this.web3.getAddress("daoFactoryAddress") ===
+            upgradeToAdCallParams.implementationAddress
+        ) {
+          res.implementation = true;
+        }
+        // initData is valid if it is empty becuse the new implementation of protocol version 1.3.0 does not need more parameters
+        if (decodedInitializeFromParams.initData?.length === 0) {
```

Comment:
this doesn't scale well, when this should be a generic implementation. For example 1.4.0 may need some initialization data

Reply:
I made a generic implementation for the active contracts, however the init data is not viable atm